src/erasure-code/isa/ErasureCodeIsa.cc

```diff
@@ -116,17 +116,49 @@ int ErasureCodeIsa::decode_chunks(const set<int> &want_to_read,
 
 // -----------------------------------------------------------------------------
 
+void
+ErasureCodeIsa::isa_xor(char **data, char **coding, int blocksize)
+{
+    // If addresses are aligned to 32 bytes, then we can use xor_gen()
+    // Otherwise, use byte_xor()
+    int i;
+    bool src_aligned = true;
+
+    for (i = 0; i < k; i++) {
+      src_aligned &= is_aligned(data[i], EC_ISA_ADDRESS_ALIGNMENT);
+    }
+
+    if (src_aligned && is_aligned(coding[0], EC_ISA_ADDRESS_ALIGNMENT)) {
+      xor_gen(k+1, blocksize, (void**) data);
+    }
+    else {
+      memcpy(coding[0], data[0], blocksize);
+      for (i = 1; i < k; i++) {
```

Comment:
Stupid question pls, why this loop starts for data at 1 instead of 0?

Reply:
Hey Jose, to calculate an xor parity we just xor each of the data chunks with each other until we're left with one chunk. On the line above this one we've copied chunk 0 from data[0] into coding[0], so then starting the loop at chunk 1 we xor data[1] against coding[0], then data[2] and so on until we're done.